### PR TITLE
Versioned gsi-qc-etl

### PR DESCRIPTION
**Pull request by slazicoicr**
Modules now load the versioned caches, which should not change with future updates of ETL. Also replaced use of strings to identify column names with the `Column` class provided by ETL.

### CODE REVIEW

File: dashi/bcl2fastq/index_summary.py [hunk 1]
```diff
@@ -141,11 +143,11 @@ def create_unknown_index_bar(pruned):
               """
 
     data_unknown = []
-    for lane, d in pruned.groupby("LaneNumber"):
+    for lane, d in pruned.groupby(un_col.LaneNumber):
         data_unknown.append(
             {
                 "x": list(d["index"]),
```
Comment (apmasell) [inline dashi/bcl2fastq/index_summary.py]: Why is this one not usinga  constant?

File: dashi/bcl2fastq/index_summary.py [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
 import dash_core_components as dcc
 import dash_html_components as html
 import dash.dependencies as dep
-import pandas
 
-index = gsiqcetl.bcl2fastq.parse.load_cache(
-    gsiqcetl.bcl2fastq.parse.CACHENAME.SAMPLES, "./data/bcl2fastq_cache.hd5"
+
+index = gsiqcetl.bcl2fastq.cache.load_known_samples_cache(
+    "v1", "./data/bcl2fastq_known_samples_cache.hd5"
```
Comment (callunity) [inline dashi/bcl2fastq/index_summary.py]: I see that some of the caches are "v1" and others are "v2". It's not clear from the gsi-qc-etl repository which modules have multiple versions, and what the differences between versions are. Could that be added somewhere, either in the README or linked prominently from the README?

File: dashi/bcl2fastq/index_summary.py (hunk 1 above)
Comment (slazicoicr) [reply dashi/bcl2fastq/index_summary.py]: I have made columns defined within a module module-level constants.

File: dashi/bamqc/gbovertime.py [hunk 3]
```diff
@@ -4,14 +4,17 @@
...
 import plotly.graph_objs as go
 
-from app import app
+import gsiqcetl.bamqc.cache
 
-bamqc = pd.read_hdf("./data/bamqc_cache.hd5", "bamqc")
-bamqc["total bases"] = bamqc["total reads"] * bamqc["average read length"]
-bamqc["run date"] = pd.to_datetime(bamqc["run name"].str[0:6], format="%y%m%d")
+
+bamqc = gsiqcetl.bamqc.cache.load_cache("v1", "./data/bamqc_cache.hd5")
```
[the remark this answers is not in the document]
Comment (slazicoicr) [reply dashi/bamqc/gbovertime.py]: The README has been updated to state that cache location needs to be provided by the environmental variable. Cache loading no longer needs to provide a file path.

File: dashi/poolqc/pooling_qc_sample.py [hunk 4]
```diff
@@ -1,5 +1,4 @@
-import gsiqcetl.bcl2fastq.parse
-import gsiqcetl.bcl2fastq.utility
+# TODO: Replace string column names with ETL Column class constants
```
[the remark this answers is not in the document]
Comment (slazicoicr) [reply dashi/poolqc/pooling_qc_sample.py]: I wrote this TODO for later work. `pooling_qc_sample` needs a lot of work and there are higher priority issues that I am working on now. I will properly fix up this module when I refactor it.

File: dashi/bcl2fastq/index_summary.py (hunk 2 above)
Comment (slazicoicr) [reply dashi/bcl2fastq/index_summary.py]: I have replaced the string by the gsi-qc-etl provided constants. Documentation of each schema is currently in the `columns` modules of gsi-qc-etl.

File: README.md [hunk 5]
```diff
@@ -8,6 +8,11 @@ Interactive visualization of OICR Genomics QC data
 
 `pip install git+https://github.com/oicr-gsi/dashi`
 
+Use the environmental variable to specify where the gsi-qc-etl cache folder
+resides:
+
+`export GSI_ETL_CACHE_DIR=/path/to/etl/cale/folder`
```
Comment (callunity) [inline README.md]: `cale` -> `cache`

**Comment by slazicoicr** [comment]
@callunity What error do you get?

**Comment by callunity** [comment]
> @callunity What error do you get?

It's an error with my installation as opposed to with the Dashi code. I ended up having to torch my Dashi installation and I re-installed it with `pip install -e git+https://github.com/oicr-gsi/dashi#egg=dashi --src DICT` and set the `GSI_ETL_CACHE_DIR` variable. When I try to run `python3 index.py` I get `ModuleNotFoundError: No module named dash_html_components` so it looks like something didn't install properly on my end.

**Pull request merged**